DDDEastAnglia.Tests/Helpers/LoginMethods/LoginMethodIconProviderTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using DDDEastAnglia.Helpers.AppSettings;
+using DDDEastAnglia.Helpers.LoginMethods;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace DDDEastAnglia.Tests.Helpers.LoginMethods
+{
+    [TestFixture]
+    public sealed class LoginMethodIconProviderTests
+    {
+        [Test]
+        public void GetIcon_ReturnsTheConfiguredIconForOAuthProvider()
+        {
+            var appSettingsProvider = Substitute.For<IAppSettingsProvider>();
+            appSettingsProvider.GetSetting(LoginMethodIconProvider.AppSettingsKeyPrefix + "test").Returns("test-icon");
```

Comment:
I agree that doesn't feel right, but the reality is:

* We make a change
* ...
* It gets pushed to production
* The site is recompiled on deployment
* The changes take effect

So I don't think the "changing it without recompiling" argument holds much sway here. Perhaps the issue is more one of incorrect decomposition, i.e. the components are at the wrong granularity and so the logic is in the wrong place?